internal/transport/http_util_test.go

```diff
@@ -22,28 +22,46 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"math"
 	"net"
 	"reflect"
 	"testing"
 	"time"
 )
 
-func (s) TestTimeoutDecode(t *testing.T) {
+func (s) TestDecodeTimeout(t *testing.T) {
 	for _, test := range []struct {
 		// input
 		s string
 		// output
 		d   time.Duration
 		err error
 	}{
+		{"0S", 0, nil},
+		{"00000000S", 0, nil},
+		{"00000001n", time.Nanosecond, nil},
+		{"10u", time.Microsecond * 10, nil},
+		{"00000010m", time.Millisecond * 10, nil},
 		{"1234S", time.Second * 1234, nil},
+		{"00000001M", time.Minute, nil},
+		{"09999999S", time.Second * 9999999, nil},
+		{"99999999S", time.Second * 99999999, nil},
+		{"99999999M", time.Minute * 99999999, nil},
+		{"2562047H", time.Hour * 2562047, nil},
+		{"2562048H", time.Duration(math.MaxInt64), nil},
+		{"99999999H", time.Duration(math.MaxInt64), nil},
+		{"-1S", 0, fmt.Errorf("strconv.ParseUint: parsing %q: invalid syntax", "-1")},
```

Comment:
This error string is not part of gRPC and can cause the test to break when run with a different version of golang. You can instead use cmp.Equal() with cmpopts.EquateErrors(). You can use `cmpopts.AnyError` as the expected value.

https://google.github.io/styleguide/go/decisions#test-error-semantics

Reply:
Good suggestion! However, since this function is returning `fmt.Errorf`, none of the expected errors can be compared, since  I changed the error check to just be for "has an error". We could change the function to return some other typed error instead, like maybe a status error? I think the test checking for existence of an error is probably okay.